### PR TITLE
LPD-35048: Export lar file with a asset-list that contains a nonexistent group can cause exception and fail the entire export process && LPD-36092: Import of manual collection fail when asset linked is from a inexistent group 

### PR DESCRIPTION
**Pull request by gustavolimav**
Tickets: 
https://liferay.atlassian.net/browse/LPD-35048
https://liferay.atlassian.net/browse/LPD-36092

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-test/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+		Mockito.doReturn(
+			null
+		).when(
+			groupLocalService
+		).fetchGroup(
+			Mockito.anyLong()
+		);
+
+		ReflectionTestUtil.setFieldValue(
```
Comment (joshuacords) [inline modules/apps/asset/asset-list-test/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java:72]: In this test you set the ReflectionTestUtil after mocking, but in the other test you set it before Mocking - I don't know which B Chan would want but he probably won't like it.

**Review by joshuacords — CHANGES_REQUESTED** [review]
I went ahead and looked into how the basic Asset Publisher handles these 2 situations. In the case of the Dynamic Selection, the Asset Publisher does what Collections will do after this fix, so I approve what's so far happened for LPD-35048. However, if you navigate to the Collection (in Site 2) it throws a UI error - that error should either be handled by this ticket or documented as a separate bug. 

**Steps to Reproduce**:
1. Create Site 1
2. Create Site 2
3. In Site 2, create a Dynamic Collection and under Scope select Site 1
4. Delete Site 1
5. Go to the Dynamic Collection in Site 2
Error is thrown in UI.

In the case of a Manual selection, an Asset Publisher shows this message in the Widget's configuration:
image. I believe Collections should work similarly, in that Manually selected items should be remembered in the configuration and a message should be delivered to the administrator in the Collection's configuration. Ultimately Collections should be storing the item using an ERC that way the link could be restored without re-importing - but this should be done in another story.

@gustavolimav, let me know if you have any questions.

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
@gustavolimav, I've updated my review comment - I think removing unlinked assets is okay now. So my only concerns are that we document that work needs to be done for lazy instantiation and that a dynamic Collection doesn't break when you try to view it after the targeted sight is removed (could be done in another ticket.) 

Code looks good :+1:

**Comment by gustavolimav** [comment]
ci:forward:force

File: modules/apps/asset/asset-list-test/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java (hunk 1 above)
Comment (gustavolimav) [reply modules/apps/asset/asset-list-test/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java:72]: lets see what brian will say about it.

**Comment by gustavolimav** [comment]
Hey @joshuacords, as soon as this ticket is merged I will file a new ticket with your finds. Thanks for the detailed review and for register all your finding here.

**Comment by gustavolimav** [comment]
ci:forward

**Comment by gustavolimav** [comment]
ci:forward:force

**Comment by gustavolimav** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/154261

**Comment by gustavolimav** [comment]
https://liferay.atlassian.net/browse/LPD-36773
https://liferay.atlassian.net/browse/LPD-36771

**Pull request closed without merging**